railties/test/application/rake/multi_dbs_test.rb

```diff
@@ -299,6 +299,39 @@ class TwoMigration < ActiveRecord::Migration::Current
         db_migrate_and_schema_cache_dump_and_schema_cache_clear
       end
 
+      test "db:abort_if_pending_migrations works on all databases" do
+        require "#{app_path}/config/environment"
+
+        app_file "db/animals_migrate/02_two_migration.rb", <<-MIGRATION
+          class TwoMigration < ActiveRecord::Migration::Current
+          end
+        MIGRATION
+
+        begin
+          rails("db:abort_if_pending_migrations")
+          fail "rails db:abort_if_pending_migrations should have aborted"
+        rescue RuntimeError => e
+          assert_match(/You have 1 pending migration/, e.message)
+        end
```

Comment:
Is there another way to check pending migrations? This feels a rough way to test that pending migrations work correctly. 1) I don't like the usage of `fail` and 2) I don't like recusing the error.

Reply:
Sorry about that, I wasn't aware of the `allow_failure:` param on `rails(...)`. I've updated and squashed appropriately.